app/src/main/java/fr/free/nrw/commons/media/MediaDetailPagerFragment.java

```diff
@@ -78,22 +78,31 @@ public ArrayList<Integer> getRemovedItems() {
         return removedItems;
     }
 
-    private MediaDetailPagerFragment() {}; // Constructor calls made to be explicit
-    @SuppressLint("ValidFragment")
-    public MediaDetailPagerFragment(Boolean editable, boolean isFeaturedImage) {
-        this.editable = editable;
-        this.isFeaturedImage = isFeaturedImage;
-        isFromFeaturedRootFragment = false;
+
+    public static MediaDetailPagerFragment newInstance(boolean editable, boolean isFeaturedImage) {
```

Comment:
Would you mind adding a Javadoc comment to these two methods, explaining the reason of this construction? It will help many readers. Thanks! :-)

Reply:
Sure. I will add the comments and fix the failed unit test shortly

Reply:
@nicolas-raoul  

1. Added method level comment. 
2. Cleaned up code, removed overloaded newInstance which was called only in unit test. 
3. Fixed failing unit test. All the test case are passing now.

![image](https://github.com/commons-app/apps-android-commons/assets/20368432/1e805057-c807-4323-83e9-cc24b0c24677)